cowprotocol/cowswap
# feat(hackathon-surplus): show surplus modal when confirmation modal is closed

# Summary

Continuing https://github.com/cowprotocol/cowswap/pull/2752

Trigger surplus modal when order is fulfilled even if confirmation modal is closed

# ⚠️ Notes ⚠️
- Styling not included, coming on https://github.com/cowprotocol/cowswap/pull/2767
- Only tested SWAP, but should NOT trigger for LIMIT
- I have not tested multiple orders fulfilling at the same time, might be an interesting edge case to check

# To Test

1. Place order
2. Close confirmation modal
* When fulfilled, modal should show up
3. Place order
4. Keep modal open
* Surplus should be displayed
5. Close modal
* No other modals should be "hidden" underneath the first one
